docs/ARCHITECTURE.md

```diff
@@ -217,12 +321,15 @@ classDiagram
 direction LR
     namespace Ai {
         class AiClient {
-            +prompt(?string $text) PromptBuilder$
-            +message(?string $text) MessageBuilder$
+            +prompt(string|Message ...$text) PromptBuilder$
+            +message(string $text) MessageBuilder$
```

Comment:
Regardless of what we decide for `prompt()` (see https://github.com/WordPress/php-ai-client/pull/12#discussion_r2205325982), I think for `message()` the parameter needs to remain optional, as not every message needs to start with a text, plus there's `withText()` as an alternative.
```suggestion
            +message(?string $text) MessageBuilder$
```